flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskExecutorGateway.java

```diff
@@ -129,4 +130,11 @@
 	 * @return Future acknowledge if the task is successfully canceled
 	 */
 	Future<Acknowledge> cancelTask(ExecutionAttemptID executionAttemptID, @RpcTimeout Time timeout);
+
+	/**
+	 *  request heartbeat from the resource manager
+	 *
+	 * @param resourceID unique id of the resource manager
+	 */
+	void requestHeartbeatFromResourceManager(ResourceID resourceID);
```

Comment:
Same applies here for the naming as well as the outsourcing of this method in a dedicated heartbeat interface.